lib/src/http/managers/interaction_manager.dart

```diff
@@ -48,11 +49,18 @@ class InteractionManager {
       raw['message'],
       (Map<String, Object?> raw) => (client.channels[channelId!] as PartialTextChannel).messages.parse(raw, guildId: guildId),
     );
-    final appPermissions = maybeParse(raw['app_permissions'], (String raw) => Permissions(int.parse(raw)));
+    final appPermissions = Permissions(int.parse(raw['app_permissions'] as String));
     final locale = maybeParse(raw['locale'], Locale.parse);
     final guildLocale = maybeParse(raw['guild_locale'], Locale.parse);
     final entitlements = parseMany(raw['entitlements'] as List, client.applications[applicationId].entitlements.parse);
 
+    final authorizingIntegrationOwners = maybeParse(
+            raw['authorizing_integration_owners'],
+            (map) =>
+                (map as Map).cast<String, Object>().map((key, value) => MapEntry(ApplicationIntegrationType.parse(int.parse(key)), Snowflake.parse(value)))) ??
+        {};
```

Comment:
```suggestion
    final authorizingIntegrationOwners = maybeParse(
            raw['authorizing_integration_owners'],
            (Map<String, Object?> map) => {
              for (final MapEntry(:key, :value) in map.entries)
                ApplicationIntegrationType.parse(int.parse(key)): Snowflake.parse(value),
            },
          ) ??
        {};
```

Also, the `authorizingIntegrationOwners` should be nullable instead of having `?? {}` if we can't parse `authorizing_integration_owners`.